### PR TITLE
More descriptive error message when salt is empty

### PR DESCRIPTION
**Pull request by antja0**
Please consider this, caused me unnecessary trouble when debugging.

See the comment in issue #52 .

### CODE REVIEW

File: src/BCrypt.Net/BCrypt.cs [hunk 1]
```diff
@@ -606,7 +606,7 @@ public static string HashPassword(string inputKey, string salt, bool enhancedEnt
 
             if (string.IsNullOrEmpty(salt))
```
Comment (snekbaev) [inline src/BCrypt.Net/BCrypt.cs]: @ChrisMcKee should the check be `IsNullOrWhiteSpace` in that case?

Comment (ChrisMcKee) [reply src/BCrypt.Net/BCrypt.cs]: IsNullOrWhiteSpace was a convenience method added in .net 4; we still target 2.0 upwards / not much to gain from adding the extra checks as they salt would still fail at the next step.
The checks at the top are only really there as a basic guard statement.

**Comment by ChrisMcKee** [comment]
No problems with that. Thanks

**Comment by ChrisMcKee** [comment]
Generally you shouldn't hit this error unless you're using the underlying methods directly which is discouraged unless absolutely necessary. i.e. we use them to validate that vectors generated in other libraries validate and continue to validate over time.

**Pull request merged**